core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts

```diff
@@ -0,0 +1,182 @@
+  let rating: ExplorationRatings;
+  let storyId = 'storyId';
+  let nodeId = 'nodeId';
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        LessonInformationCardModalComponent,
+        MockTranslatePipe,
+        MockTruncteAndCapitalizePipe,
+      ],
+      providers: [
+        NgbActiveModal,
+        PlayerTranscriptService,
+        ExplorationEngineService,
+        StoryViewerBackendApiService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LessonInformationCardModalComponent);
+    componentInstance = fixture.componentInstance;
+
+    componentInstance.expInfo = {
+      category: '',
+      community_owned: true,
+      activity_type: '',
+      last_updated_msec: 0,
+      ratings: rating,
+      id: expId,
+      created_on_msec: 2,
+      human_readable_contributors_summary: {
+        'contributer 1': {
+          num_commits: 2
+        },
+        'contributer 2': {
+          num_commits: 2
+        }
+      },
+      language_code: '',
+      num_views: 3,
+      objective: expDesc,
+      status: 'private',
+      tags: [],
+      thumbnail_bg_color: '',
+      thumbnail_icon_url: '',
+      title: expTitle
+    };
+
+    i18nLanguageCodeService = TestBed.inject(I18nLanguageCodeService);
+    urlService = TestBed.inject(UrlService);
+    storyViewerBackendApiService = TestBed.inject(StoryViewerBackendApiService);
+
+    spyOn(i18nLanguageCodeService, 'isHackyTranslationAvailable')
+      .and.returnValue(true);
+    spyOn(i18nLanguageCodeService, 'isCurrentLanguageEnglish')
+      .and.returnValue(false);
+  });
+
+  it('should intialize the component and set values', fakeAsync(() => {
+    componentInstance.ngOnInit();
+    expect(componentInstance.explorationId).toEqual(expId);
+    expect(componentInstance.expTitle).toEqual(expTitle);
+    expect(componentInstance.expDesc).toEqual(expDesc);
+
+    expect(componentInstance.expTitleTranslationKey).toEqual(
+      'I18N_EXPLORATION_expId_TITLE');
+    expect(componentInstance.expDescTranslationKey).toEqual(
+      'I18N_EXPLORATION_expId_DESCRIPTION');
+
+    // Translation is only displayed if the language is not English
+    // and it's hacky translation is available.
+    let hackyExpTitleTranslationIsDisplayed =
+      componentInstance.isHackyExpTitleTranslationDisplayed();
+    expect(hackyExpTitleTranslationIsDisplayed).toBe(true);
+    let hackyExpDescTranslationIsDisplayed =
+      componentInstance.isHackyExpDescTranslationDisplayed();
+    expect(hackyExpDescTranslationIsDisplayed).toBe(true);
+    let hackyStoryTitleTranslationIsDisplayed =
+      componentInstance.isHackyStoryTitleTranslationDisplayed();
+    expect(hackyStoryTitleTranslationIsDisplayed).toBe(true);
+  }));
+
+  it('should set hasStoryTitle true when' +
+      'storyId is not undefined',
```

Comment:
Can we rewrite the test description so it does not talk about variable names -- it should be from a users perspective? See https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly